src/register.cc

```diff
@@ -161,6 +162,7 @@ ncclResult_t ncclCommRegister_impl(const ncclComm_t comm, void* buff, size_t siz
   NCCLCHECK(CommCheck(comm, "ncclCommRegister", "comm"));
   if (comm->checkPointers) NCCLCHECK(CudaPtrCheck(buff, comm, "buff", "ncclCommRegister"));
   #ifdef ENABLE_MSCCLPP
+  if (rcclParamMscclppEnabled()) {
```

Comment:
Is `comm->mscclppCompatible` a valid indicator instead? I'm not sure, because it is set in `ncclCommInitRankFunc`, but it might be. Note that it also considers the gcnArch.

Reply:
I think the `comm->mscclCompatible` check below should stay, but this could be changed to `comm->mscclppCompatible`.
```suggestion
  if (comm->mscclppCompatible) {
```